src/services/query-options.js

```diff
@@ -0,0 +1,248 @@
+    this._options = options;
+
+    // Used to compute relations that will go in the final 'include'
+    this._requestedFields = new Set();
+    this._neededFields = new Set();
+    this._scopes = []; // @see sequelizeScopes getter
+
+    // Other sequelize params
+    this._where = [];
+    this._order = [];
+    this._offset = undefined;
+    this._limit = undefined;
+
+    if (this._options.includeRelations) {
+      _.values(this._model.associations)
+        .filter((association) => ['HasOne', 'BelongsTo'].includes(association.associationType))
+        .forEach((association) => this._requestedFields.add(association.associationAccessor));
+    }
+  }
+
+  /**
+   * Add the required includes from a list of field names.
+   * @param {string[]} fields Fields of HasOne and BelongTo relations are
+   *  accepted (ie. 'book.name').
+   */
+  async requireFields(fields) {
+    if (fields) {
+      fields.forEach(this._requestedFields.add, this._requestedFields);
+    }
+  }
+
+  /**
+   * Filter resulting query set with packed primary ids.
+   * This works both for normal collection, and those which use composite primary keys.
+   * @param {string[]} recordIds Packed record ids
+   */
+  async filterByIds(recordIds) {
+    this._where.push(new PrimaryKeysManager(this._model).getRecordsConditions(recordIds));
+  }
+
+  /**
+   * Apply condition tree to those query options (scopes, user filters, charts, ...)
+   * @param {*} filters standard forest filters
+   * @param {*} timezone timezone of the user (required if filtering on dates)
+   */
+  async filterByConditionTree(filters, timezone) {
+    const filterParser = new FiltersParser(this._schema, timezone, { Sequelize: this._Sequelize });
+
+    if (filters) {
+      const whereClause = await filterParser.perform(filters);
+      this._where.push(whereClause);
+
+      const associations = await filterParser.getAssociations(filters);
+      associations.forEach(this._neededFields.add, this._neededFields);
+    }
+  }
+
+  /**
+   * Retrict rows to those matching a search string
+   * @param {string} search search string
+   * @param {boolean|string} searchExtended if truthy, enable search in relations
+   */
+  async search(search, searchExtended) {
+    if (!search) return [];
+
+    const options = { Sequelize: this._Sequelize };
+    const fieldNames = this._requestedFields.size ? [...this._requestedFields] : null;
+    const helper = new SearchBuilder(this._model, options, { search, searchExtended }, fieldNames);
+
+    const conditions = helper.performWithSmartFields(this._options.tableAlias);
+    if (conditions) {
+      this._where.push(conditions);
+    } else {
+      this._where.push(this._Sequelize.literal('(0=1)'));
+    }
+
+    return helper.getFieldsSearched();
+  }
+
+  /**
+   * Apply a forestadmin segment
+   * @param {string} name name of the segment (from the querystring)
+   * @param {string} segmentQuery SQL query of the segment (also from querystring)
```

Comment:
segmentQuery ? There is no such params 🤔

Reply:
Sadly, there is such a param, and it's a SQL injection

https://github.com/ForestAdmin/forest-express-sequelize/blob/master/src/services/resources-getter.js#L95-L98

Reply:
then it shoul be a single param called nameOrSegmentQuery ?